src/executables/csql.c

```diff
@@ -3330,3 +3358,129 @@ get_host_ip (unsigned char *ip_addr)
 
   return 0;
 }
+
+static int
+csql_connect (char *argument, CSQL_ARGUMENT * csql_arg)
+{
+  /*dbname can be stored DB name+ @ + remote hostname + \0 */
+  char buf[DB_MAX_USER_LENGTH + MAXLOGNAME + CUB_MAXHOSTNAMELEN + 1];
+  const char *delim = " \n";
+  char *save_ptr_strtok = NULL;
+  const char *db_name_ptr = NULL;
+  char *user_name_ptr = NULL;
+  char *host_name_ptr = NULL;
+  char *p = NULL;
+  const char *err_msg;
+  CSQL_ARGUMENT csql_new_arg;
+
+  if (argument == NULL)
+    {
+      err_msg = (*csql_get_message) (CSQL_MSG_TOO_FEW_ARGS);
+      fprintf (stderr, err_msg);
+      return DO_CMD_SUCCESS;
+    }
+
+  /*verify @hostname */
+  if ((host_name_ptr = strchr (argument, '@')) != NULL && (*(host_name_ptr + 1) == '\0'))
+    {
+      csql_Error_code = CSQL_ERR_SQL_ERROR;
+      return DO_CMD_FAILURE;
+    }
+
+  strncpy (buf, argument, sizeof (argument));
```

Comment:
Did you test the code after fixing it?
This is the code with the error.

Reply:
Right. I will fix it. Thank you.

Reply:
I think
`  strcpy (buf, argument);`

Reply:
@airnet73 recommend,

`strncpy (buf, argument, sizeof(buf) - 1);`